File: app/src/main/java/com/boardgamegeek/service/CollectionStatusUploadTask.java
package com.boardgamegeek.service;

import android.content.ContentValues;

import com.boardgamegeek.provider.BggContract.Collection;
import com.boardgamegeek.service.model.CollectionItem;

import okhttp3.FormBody;
import okhttp3.OkHttpClient;
import timber.log.Timber;

public class CollectionStatusUploadTask extends CollectionUploadTask {
	//	<div class='owned'>Owned</div>
	//	<div class='wishlist'>Wishlist(3)
	//	<br>&nbsp;(Like&nbsp;to&nbsp;have)
	//	</div>

	public CollectionStatusUploadTask(OkHttpClient client) {
		super(client);
	}

	@Override
	public String getTimestampColumn() {
		return Collection.STATUS_DIRTY_TIMESTAMP;
	}

	@Override
	public boolean isDirty() {
		return collectionItem.getStatusTimestamp() > 0;
	}

	@Override
	protected FormBody createForm(CollectionItem collectionItem) {
		return createFormBuilder()
			.add("fieldname", "status")
			.add("own", collectionItem.owned() ? "1" : "0")
			.add("prevown", collectionItem.previouslyOwned() ? "1" : "0")
			.add("fortrade", collectionItem.forTrade() ? "1" : "0")
			.add("want", collectionItem.wantInTrade() ? "1" : "0")
			.add("wanttobuy", collectionItem.wantToBuy() ? "1" : "0")
			.add("wanttoplay", collectionItem.wantToPlay() ? "1" : "0")
			.add("preordered", collectionItem.preordered() ? "1" : "0")
			.add("wishlist", collectionItem.wishlist() ? "1" : "0")
			.add("wishlistpriority", String.valueOf(collectionItem.wishlistPriority()))
			.build();
	}

	@Override
	protected void saveContent(String content) {
		Timber.d(content);
	}

	@Override
	public void appendContentValues(ContentValues contentValues) {
		contentValues.put(Collection.STATUS_DIRTY_TIMESTAMP, 0);
	}
}

File: app/src/main/java/com/boardgamegeek/service/CollectionStatusUploadTask.kt
package com.boardgamegeek.service

import android.content.ContentValues
import com.boardgamegeek.provider.BggContract
import okhttp3.FormBody
import okhttp3.OkHttpClient
import timber.log.Timber

class CollectionStatusUploadTask(client: OkHttpClient) : CollectionUploadTask(client) {
    //	<div class='owned'>Owned</div>
    //	<div class='wishlist'>Wishlist(3)
    //	<br>&nbsp;(Like&nbsp;to&nbsp;have)
    //	</div>
    override val timestampColumn = BggContract.Collection.STATUS_DIRTY_TIMESTAMP

    override val isDirty = collectionItem.statusTimestamp > 0

    override fun createForm(): FormBody {
        @Suppress("SpellCheckingInspection")
        return createFormBuilder()
                .add("fieldname", "status")
                .add("own", if (collectionItem.owned) "1" else "0")
                .add("prevowned", if (collectionItem.previouslyOwned) "1" else "0")
                .add("fortrade", if (collectionItem.forTrade) "1" else "0")
                .add("want", if (collectionItem.wantInTrade) "1" else "0")
                .add("wanttobuy", if (collectionItem.wantToBuy) "1" else "0")
                .add("wanttoplay", if (collectionItem.wantToPlay) "1" else "0")
                .add("preordered", if (collectionItem.preordered) "1" else "0")
                .add("wishlist", if (collectionItem.wishlist) "1" else "0")
                .add("wishlistpriority", collectionItem.wishlistPriority.toString())
                .build()
    }

    override fun saveContent(content: String) {
        Timber.d(content)
    }

    override fun appendContentValues(contentValues: ContentValues) {
        contentValues.put(BggContract.Collection.STATUS_DIRTY_TIMESTAMP, 0)
    }
}